src/domain/entities/Action.ts

```diff
@@ -87,3 +89,28 @@ export const defaultAction: PartialAction = {
     dhisAuthorities: [],
     disabled: false,
 };
+
+export const getPageActions = (
+    isRoot: boolean,
+    showAllActions: boolean,
+    actions: Action[],
+    user: User,
+    currentPageActions: Action[]
+) => {
+    if (isRoot && showAllActions) return actions.map(({ id }) => id);
+    else if (user)
+        return currentPageActions
+            .filter(action => {
+                const actionUsers = action.userAccesses?.map(userAccess => userAccess.id);
+                const actionUserGroups = action.userGroupAccesses?.map(userGroupAccess => userGroupAccess.id);
+                const userGroupIds = user.userGroups?.map(userGroup => userGroup.id);
+
+                const hasUserAccess = actionUsers.includes(user.id);
+                const hasUserGroupAccess = _.intersection(actionUserGroups, userGroupIds).length > 0;
+                const hasPublicAccess = action.publicAccess !== "--------";
+
+                return hasUserAccess || hasUserGroupAccess || hasPublicAccess;
+            })
+            .map(({ id }) => id);
+    else return [];
```

Comment:
[minor, subjective] Unless for some specific reason,  prefer if/else if/else blocks to have `{}` brackets to improve readability.